Sources/Commands/SwiftBuildTool.swift

```diff
@@ -212,43 +86,76 @@ public class SwiftBuildTool: SwiftTool<BuildToolMode, BuildToolOptions> {
     }
 }
 
-extension Build.Configuration {
-    fileprivate init(_ rawValue: String?) throws {
-        switch rawValue?.lowercased() {
-        case "debug"?:
+public class BuildToolOptions: ToolOptions {
+    /// Returns the mode in which the build tool should run.
+    func mode() throws -> BuildToolMode {
+        // If we got version option, just print the version and exit.
+        if printVersion {
+            return .version
+        }
+        // FIXME: We should change the behavior to clean and then build, not clean and exit.
```

Comment:
Why? This isn't typically what clean does in command line build tools... if we want to do this we should probably call it something else, like `swift build --from-scratch`.